src/bp/ui-studio/src/web/components/Shared/ElementsList/InputElement.jsx

```diff
@@ -0,0 +1,50 @@
+import React from 'react'
+import classnames from 'classnames'
+import style from './style.scss'
+
+export class InputElement extends React.Component {
+  elementInputRef = React.createRef()
+
+  state = {
+    error: undefined
+  }
+
+  onEnter = value => {
+    this.props.onEnter(value)
+  }
+
+  handleOnEnter = event => {
+    const inputValue = event.target.value
+    const enterKeyPressed = event.keyCode === 13
+    const inputNotEmpty = inputValue !== ''
+    const elementUnique = !this.props.elements.includes(inputValue)
+
+    if (enterKeyPressed) {
+      if (inputNotEmpty && elementUnique) {
+        this.setState({ error: undefined }, () => this.onEnter(inputValue))
+        if (this.props.eraseValue) {
+          this.elementInputRef.current.value = ''
```

Comment:
Input field should have a value in the state and a method to update it (eg= onChange=this.handleInputChanged and value=this.state.inputValue). ref should be mostly used for special cases, like focus

Reply:
I understand. I this case however, we don't have to maintain the state of the input because the value is discarded as soon as we press enter. I think the fact the we don't have to update a state every time the input value change far outperforms a ref.